cpp/arcticdb/storage/s3/s3_storage.cpp

```diff
@@ -42,6 +86,7 @@ S3Storage::S3Storage(const LibraryPath &library_path, OpenMode mode, const Confi
 
     if (creds.GetAWSAccessKeyId() == USE_AWS_CRED_PROVIDERS_TOKEN && creds.GetAWSSecretKey() == USE_AWS_CRED_PROVIDERS_TOKEN){
         ARCTICDB_RUNTIME_DEBUG(log::storage(), "Using AWS auth mechanisms");
+        check_ec2_metadata_endpoint(*s3_api_);
```

Comment:
Why does this change necessitate this line?

Reply:
What are you referring to by "this change"?

Reply:
This PR. Checking the metadata endpoint seems unrelated?

Reply:
The metadata endpoint is used to fetch the machine identity, which is one of the AWS authentication methods.

We added logic to disable the metadata endpoint if it is not reachable because it's causing that delay with connection creation internally.

However, if the user is running in AWS and uses machine identity, then s/he might want to know we have disabled it. Otherwise, the authentication failure might become a mystery.